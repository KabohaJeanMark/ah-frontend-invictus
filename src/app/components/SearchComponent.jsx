import React from 'react';
import NavbarComponent from '../components/Navbar';
import { Link } from "react-router-dom";
import '../../assets/styles/articles.scss';

const SearchComponent = (props) => {
  const { handleSearch, handleChange, search } = props;
  const articles = props.data.article;
  return (
    <div>
      <NavbarComponent />
      <div className="row"><div className="col-md-1" />
        <div className="col-md-10">
          <form onSubmit={handleSearch}>
            <div className="input-group mb-3">
              <input type="text" name="search" onChange={handleChange}
                value={search} className="form-control" placeholder="type something..."
                aria-label="type something..." aria-describedby="button-addon2" /><div>
                <select className="form-control" name="filter" onChange={handleChange} id="filterSelectID">
                  <option value="none">none</option>
                  <option value="tag">Tag</option>
                  <option value="author">Author</option>
                  <option value="title">Title</option></select></div>
              <label className="text-white" htmlFor="filterSelectID">Filter by</label>
              <div className="input-group-append">
                <button className="btn btn-outline-secondary" type="submit" id="button-addon2">Search</button>
              </div></div></form><div>
            {
              (articles === undefined) || (articles.length === 0) ? "No Articles to display" : (articles.map(article => (
                <Link key={article.id} className="card-article-link" to={`/articles/${article.id}`}>
                  <div className="container">
                    <div className="row">
                      <div className="card-article" style={{ marginLeft: "25%" }}>
                        <img className="card-image" alt="" src="https://picsum.photos/600/350" />
                        <div className="card-body">
                          <div><p className="card-publish-date">Published On:
                            {article.created_at.substring(0, 10)}</p></div>
                          <h4 className="card-title">{article.title}</h4><p className="card-text">{article.description}</p>
                          <div className="card-body-footer">
                            <p className="card-footer-text">@{article.author}</p></div>
                          <div className="all-tags"><hr />
                            <ul>{article.tagList.map(tag => <li key={tag} className="tag">{tag}</li>)}</ul></div>
                        </div></div></div></div>
                </Link>)))
            }
          </div></div><div className="col-md-1" /></div></div>
  );
};

export default SearchComponent;
