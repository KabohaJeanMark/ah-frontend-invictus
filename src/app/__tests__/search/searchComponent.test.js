import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchComponent from '../../components/SearchComponent';

describe('<SearchComponent />', () => {
  const props = {
    data: {
      article:
        [{
          id: 1,
          created_at: 'Test',
          title: 'Test',
          description: 'Test',
          author: 'Test',
          tagList: ["edna"],
          read_time: 'Test'
        }]
    }
  };
  let wrapper;
  it('it renders search without crashing', () => {
    wrapper = shallow(<SearchComponent {...props}/>);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
