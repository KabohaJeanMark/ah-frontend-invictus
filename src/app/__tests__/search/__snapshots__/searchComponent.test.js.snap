// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchComponent /> matches snapshot 1`] = `
<div>
  <Connect(NavbarComponent) />
  <div
    className="row"
  >
    <div
      className="col-md-1"
    />
    <div
      className="col-md-10"
    >
      <form>
        <div
          className="input-group mb-3"
        >
          <input
            aria-describedby="button-addon2"
            aria-label="type something..."
            className="form-control"
            name="search"
            placeholder="type something..."
            type="text"
          />
          <div>
            <select
              className="form-control"
              id="filterSelectID"
              name="filter"
            >
              <option
                value="none"
              >
                none
              </option>
              <option
                value="tag"
              >
                Tag
              </option>
              <option
                value="author"
              >
                Author
              </option>
              <option
                value="title"
              >
                Title
              </option>
            </select>
          </div>
          <label
            className="text-white"
            htmlFor="filterSelectID"
          >
            Filter by
          </label>
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-outline-secondary"
              id="button-addon2"
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </form>
      <div>
        <Link
          className="card-article-link"
          key="1"
          to="/articles/1"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="card-article"
                style={
                  Object {
                    "marginLeft": "25%",
                  }
                }
              >
                <img
                  alt=""
                  className="card-image"
                  src="https://picsum.photos/600/350"
                />
                <div
                  className="card-body"
                >
                  <div>
                    <p
                      className="card-publish-date"
                    >
                      Published On:
                      Test
                    </p>
                  </div>
                  <h4
                    className="card-title"
                  >
                    Test
                  </h4>
                  <p
                    className="card-text"
                  >
                    Test
                  </p>
                  <div
                    className="card-body-footer"
                  >
                    <p
                      className="card-footer-text"
                    >
                      @
                      Test
                    </p>
                  </div>
                  <div
                    className="all-tags"
                  >
                    <hr />
                    <ul>
                      <li
                        className="tag"
                        key="edna"
                      >
                        edna
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Link>
      </div>
    </div>
    <div
      className="col-md-1"
    />
  </div>
</div>
`;
